### PR TITLE
Smoother scale animation for dialogs and popovers

### PR DESCRIPTION
**Pull request by llorca**
Motion improvements from @pkwi and I! Check out popovers, tooltips, context menus, dialogs.

It is only visible on modern browsers -- doesn't impact IE and older ones.

### CODE REVIEW

File: packages/core/src/components/popover/_common.scss [hunk 1]
```diff
@@ -181,14 +181,34 @@ $dark-popover-background-color: $dark-gray4 !default;
...
 
-@mixin fade-transition() {
-  @include react-transition("pt-popover", (opacity: 0 1), $before: "&");
+// @mixin fade-transition() {
+//   @include react-transition(
+//     "pt-popover",
+//     (opacity: 0 1),
+//     $before: "&"
+//   );
+// }
```
Comment (giladgray) [inline packages/core/src/components/popover/_common.scss]: delete commented code

File: packages/core/src/components/popover/_popover.scss [hunk 2]
```diff
@@ -92,7 +87,7 @@ $popover-width: $pt-grid-size * 35 !default;
 }
 
 .pt-transition-container {
-  @include fade-transition();
+  // @include fade-transition();
```
Comment (giladgray) [inline packages/core/src/components/popover/_popover.scss]: remove

File: packages/core/src/components/popover/popover.tsx [hunk 3]
```diff
@@ -181,6 +181,15 @@ export interface IPopoverProps extends IOverlayableProps, IProps {
     tetherOptions?: Partial<Tether.ITetherOptions>;
 
     /**
+     * Indicates how long (in milliseconds) the tooltip's appear/disappear transition takes.
+     * This is used by React `CSSTransitionGroup` to know when a transition completes
+     * and must match the duration of the animation in CSS.
+     * Only set this prop if you override Blueprint's default transitions with new transitions of a different length.
+     * @default 300
+     */
+    transitionDuration?: number;
```
Comment (giladgray) [inline packages/core/src/components/popover/popover.tsx]: ❌ this is already defined in overlay.tsx. please update `@default` there

File: packages/core/src/components/tooltip/_tooltip.scss [hunk 4]
```diff
@@ -24,7 +24,7 @@
   // lightweight scale for tooltips
   @include react-transition(
     "pt-popover",
-    (transform: scale(0.8) scale(1)),
+    (transform: scale(1) scale(1), opacity: 0 1, filter: blur($pt-grid-size) blur(0)),
```
Comment (giladgray) [inline packages/core/src/components/tooltip/_tooltip.scss]: add note about canceling `transform` transition from popover, so we don't wonder why these are the same scales

File: packages/site-docs/src/styles/_examples.scss [hunk 5]
```diff
@@ -67,10 +67,18 @@
 .docs-overlay-example-transition {
   $overlay-example-width: $pt-grid-size * 40;
   $enter: (
-    transform: (translateY(-50vh) rotate(-10deg), translateY(0) rotate(0deg))
+    transform: (
+      translateY(-50vh) rotate(-10deg),
+      translateY(0) rotate(0deg)
+    ),
```
Comment (giladgray) [inline packages/site-docs/src/styles/_examples.scss]: undo these line breaks, they are not helpful for legibility

**Comment by llorca** [comment]
Fair, will make the changes

File: packages/core/src/components/tooltip/_tooltip.scss (hunk 4 above)
Comment (giladgray) [inline packages/core/src/components/tooltip/_tooltip.scss]: the blur doesn't work here... animation is too fast to really appreciate it.

**Comment by adidahiya** [comment]
Not even sure about the blur on tooltips... my initial reaction was that it was so subtle & fast that it fell into the valley of just-barely-perceptible and seemed like a bug rather than a feature... will play with it some more.

Also in general I'd be careful about using blur on underpowered hardware, we've already seen problems that the zoom transition causes on dialogs...

**Comment by llorca** [comment]
You're right with the tooltips, I messed something up. Fixing

**Comment by dmackerman** [comment]
Smooths out great in Safari, but it drops frames in Chrome and I'm on an i7 iMac. Just a heads up.

**Comment by adidahiya** [comment]
Yeah I also see frames being dropped intermittently in Chrome, especially on the first time I try the animation. 2.3 GHz Intel Core i7

**Comment by llorca** [comment]
I thought Chrome perf was acceptable, looks like not. Will reopen in a few months if there's progress on that front

**Comment by dimaShin** [comment]
Hi. Thank you a lot for such good components.
But Dialog animation is laggy in chrome and I want to turn it off at all. Do you have any ideas?

**Comment by llorca** [comment]
@dimaShin doesn't belong here as this PR didn't merge. File separate issue please

**Pull request closed without merging**